tendermint/src/rpc/endpoint/abci_info.rs

```diff
@@ -27,41 +27,58 @@ impl rpc::Response for Response {}
 
 /// ABCI information
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
 pub struct AbciInfo {
     /// Name of the application
     pub data: String,
 
     /// Version
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub version: Option<String>,
 
     /// Last block height
-    pub last_block_height: block::Height,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub last_block_height: Option<block::Height>,
 
     /// Last app hash for the block
     #[serde(
         serialize_with = "serialize_app_hash",
-        deserialize_with = "parse_app_hash"
+        deserialize_with = "parse_app_hash",
+        skip_serializing_if = "Option::is_none"
     )]
-    pub last_block_app_hash: Hash,
+    pub last_block_app_hash: Option<Hash>,
 }
 
 /// Parse Base64-encoded app hash
-pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Hash, D::Error>
+pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Option<Hash>, D::Error>
 where
     D: Deserializer<'de>,
 {
     let bytes = base64::decode(String::deserialize(deserializer)?.as_bytes())
         .map_err(|e| D::Error::custom(format!("{}", e)))?;
 
-    Hash::new(hash::Algorithm::Sha256, &bytes).map_err(|e| D::Error::custom(format!("{}", e)))
+    Hash::new(hash::Algorithm::Sha256, &bytes)
+        .map(Some)
+        .map_err(|e| D::Error::custom(format!("{}", e)))
```

Comment:
Maybe add a little one-liner comment above this code or in the function doc, explaining when it errs, when it returns Some and when None? What seems confusing to me: it never returns None, right? Either it errs or it returns Some(...).

Reply:
So, `Hash::new` returns a `Result<Hash,Error>`. The original function directly returned that.
The function now has to return a `Result<Option<Hash>,Error>`. As `Hash:New` cannot return `None`, using the transitive property of `Some`, our function cannot return `None` either.

So we return `Some(Hash)` or `Error`, just as the original function returned `Hash` or `Error`. The reason to make the return into an `Option` is that the whole value in the structure can be omitted.